src/Signer.php

```diff
@@ -5,11 +5,13 @@
 
 use Lcobucci\JWT\Signer\CannotSignPayload;
 use Lcobucci\JWT\Signer\Ecdsa\ConversionFailed;
-use Lcobucci\JWT\Signer\InvalidKeyProvided;
-use Lcobucci\JWT\Signer\Key;
 
 interface Signer
 {
+    public static function generateKey(): mixed;
```

Comment:
Documenting key generation is good, providing key generation is better.

Reply:
IMO this opens a can of worms - maintaining this ourselves means more security issues.

Reply:
I understand your point. Consider that:

1. It should be a one-liner: `random_bytes(32)`, `openssl_pkey_new(['private_key_type' => $this->keyType())`, `sodium_crypto_generichash_keygen()` etc
2. I would prefer to take this responsability, rather than seeing new developers still copying `uniqid()` output as valid key

Reply:
I'm on board as long as you are OK with fighting more toxic cryptographers :D

Reply:
The consumers of this method are different from the sign/verify.
Although I understand the reasons why you want to provide it, I'd postpone it and offer a different interface.